test/attr/attr_availability_narrow.swift

```diff
@@ -1,69 +1,101 @@
+  if #available(OSX 50.1, *) {
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}} {{-1:21-25=50.2}}
   }
 }
 
 func useFooAlreadyOkRange() {
-  if #available(OSX 10.51, *) {
+  if #available(OSX 51, *) {
     foo()
   }
 }
 
 func useFooUnaffectedSimilarText() {
-  if #available(iOS 10.50.10, OSX 10.50.1, *) {
-    foo() // expected-error {{'foo()' is only available in macOS 10.50.2 or newer}} {{-1:35-42=10.50.2}}
+  if #available(iOS 50.10, OSX 50.1, *) {
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}} {{-1:32-36=50.2}}
   }
 }
 
 func useFooWayOff() {
-    // expected-note@-1{{add @available attribute to enclosing global function}}
+  // expected-note@-1 2 {{add @available attribute to enclosing global function}}
   if #available(OSX 10.10, *) {
-    foo() // expected-error {{'foo()' is only available in macOS 10.50.2 or newer}}
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}}
+    // expected-note@-1{{add 'if #available' version check}}
+  }
+
+  if #available(OSX 49.0, *) {
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}}
     // expected-note@-1{{add 'if #available' version check}}
   }
 }
 
-@available(OSX 10.50, *)
+@available(OSX 50, *)
 class FooUser {
   func useFoo() {
-    foo() // expected-error {{'foo()' is only available in macOS 10.50.2 or newer}} {{-3:16-21=10.50.2}}
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}} {{-3:16-18=50.2}}
   }
 }
 
-@available(OSX, introduced: 10.50, obsoleted: 10.50.4)
+@available(OSX, introduced: 50, obsoleted: 50.4)
 class FooUser2 {
   func useFoo() {
-    foo() // expected-error {{'foo()' is only available in macOS 10.50.2 or newer}} {{-3:29-34=10.50.2}}
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}} {{-3:29-31=50.2}}
   }
 }
 
-@available(OSX, introduced: 10.50, obsoleted: 10.50.4)
+@available(OSX, introduced: 50, obsoleted: 50.4)
 @objc
 class FooUser3 : NSObject {
   func useFoo() {
-    foo() // expected-error {{'foo()' is only available in macOS 10.50.2 or newer}} {{-4:29-34=10.50.2}}
+    foo() // expected-error {{'foo()' is only available in macOS 50.2 or newer}} {{-4:29-31=50.2}}
   }
 }
 
-@available(OSX, introduced: 10.50.4)
+@available(OSX, introduced: 50.4)
 class FooUserOkRange {
   func useFoo() {
     foo()
   }
 }
+
+@available(macOS 10.50.2, *)
+func foo10() { }
+
+func useFoo10() {
+  if #available(macOS 10.50.1, *) {
+    foo10() // expected-error {{'foo10()' is only available in macOS 10.50.2 or newer}} {{-1:23-30=10.50.2}}
+  }
+}
+
+func useFoo10WayOff() {
+    // expected-note@-1 3 {{add @available attribute to enclosing global function}}
+  if #available(OSX 10.10, *) {
+    foo10() // expected-error {{'foo10()' is only available in macOS 10.50.2 or newer}}
+    // expected-note@-1{{add 'if #available' version check}}
```

Comment:
Minor: It would be nice to verify the content of the `add 'if #available'` fix-its in this test.

Reply:
These fixits are tested in Sema/availability_versions.swift and the test here is focused on the narrowing fixits not these ones. I'd prefer to keep those things independent, but let me know if you think there's something specific that belongs in this test.

Reply:
Fair enough!